src/Utils/General/DOMHelper.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace LearnosityQti\Utils;
+
+use LearnosityQti\Exceptions\MappingException;
+
+class DOMHelper
+{
+
+    public static function getDomForXml($xml)
+    {
+        $dom = new \DOMDocument();
+
+        $dom->preserveWhiteSpace = false;
+        $dom->formatOutput = false;
+        $dom->substituteEntities = false;
+
+        $isValid = $dom->loadXML($xml);
+
+        if (!$isValid) {
+            throw new MappingException('Invalid XML; Failed to parse DOM for sharedpassage content');
+        }
+
+        return $dom;
+    }
+    
+    public static function sanitizeXml($xml)
+    {
+        $xml = trim($xml);
+
+        $dom = new \DOMDocument('1.0', 'UTF-8');
+        $previousLibXmlSetting = libxml_use_internal_errors(true);
+
+        // HACK: Pass the version and encoding to prevent libxml from decoding HTML entities (esp. &amp; which libxml borks at)
+        // Only do this if it hasnt already been passed along in the xml string. Sometimes, we read from a file, and sometimes
+        // we read from a block inside another file.
+        if (strpos($xml, '<?xml ') !== false) {
+            $xml = substr($xml, strpos($xml, '>') + 1);
+        }
+        $xml = '<?xml version="1.0" encoding="UTF-8">' . "<div>$xml</div>";
+
+        $dom->loadHTML($xml, LIBXML_HTML_NODEFDTD | LIBXML_HTML_NOIMPLIED);
+        $xml = $dom->saveXML($this->getFragmentWrapperDocumentElementForDom($dom));
```

Comment:
You haven't moved this `getFragmentWrapperDocumentElementForDom()` method into this class from https://github.com/Learnosity/learnosity-qti/pull/55/files#diff-b3f82e7247236692ccc3e7fa383883e2R238. This will break without it.

Reply:
Done